libs/parallelism/futures/tests/unit/direct_scoped_execution.cpp

```diff
@@ -0,0 +1,95 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2011 Bryce Lelbach
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+// This is example is equivalent to fibonacci.cpp, except that this example does
+// not use actions (only plain functions). Many more variations are found in
+// fibonacci_futures.cpp. This example is mainly intended to demonstrate async,
+// futures and get for the documentation.
+
+#include <hpx/init.hpp>
+#include <hpx/local/chrono.hpp>
+#include <hpx/local/execution.hpp>
+#include <hpx/local/future.hpp>
+#include <hpx/modules/format.hpp>
+#include <hpx/modules/testing.hpp>
+
+#include <cstdint>
+#include <iostream>
+#include <string>
+#include <vector>
+
+bool use_scoping = true;
+
+///////////////////////////////////////////////////////////////////////////////
+std::uint64_t fibonacci(std::uint64_t n)
+{
+    if (n < 2)
+        return n;
+
+    hpx::threads::thread_schedule_hint hint;
+    hint.runs_as_child = use_scoping;
+
+    auto exec = hpx::execution::experimental::with_hint(
+        hpx::execution::parallel_executor{}, hint);
+
+    hpx::future<std::uint64_t> n1 = hpx::async(exec, fibonacci, n - 1);
+    std::uint64_t n2 = fibonacci(n - 2);
+
+    // wait for the Futures to return their values
+    return n1.get() + n2;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+int hpx_main(hpx::program_options::variables_map& vm)
+{
+    hpx::threads::add_scheduler_mode(hpx::threads::policies::fast_idle_mode);
```

Comment:
Is this necessary for the scoped execution to work? Does it have an effect on the performance?

Reply:
Right, I'll remove this. I copied the code from another example...